ExampleMod/Items/Weapons/ExampleSentryItem.cs

```diff
@@ -0,0 +1,42 @@
+using ExampleMod.Projectiles.Minions;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Items.Weapons
+{
+	public class ExampleSentryItem : ModItem
+	{
+		public override void SetStaticDefaults()
+		{
+			Tooltip.SetDefault("Summons an example sentry to fight for you");
+			ItemID.Sets.GamepadWholeScreenUseRange[item.type] = true;
+			ItemID.Sets.LockOnIgnoresCollision[item.type] = true;
+		}
+
+		public override void SetDefaults()
+		{
+			item.damage = 50;
+			item.sentry = true;
+			item.mana = 10; //How much mana this weapon takes to use.
+			item.width = 26; //Item width hitbox.
+			item.height = 28; //Item height hitbox.
+			item.useTime = 36;
+			item.useAnimation = 36;
+			item.useStyle = ItemUseStyleID.SwingThrow;
+			item.noMelee = true; //Restricts this weapon dealing melee damage.
+			item.knockBack = 3;
+			item.value = Item.buyPrice(0, 30, 0, 0); //How much this item is sold for.
+			item.rare = ItemRarityID.Cyan;
+			item.UseSound = SoundID.Item83;
+			item.shoot = ModContent.ProjectileType<ExampleSentry>();
+		}
+
+		public override bool Shoot(Player player, ref Vector2 position, ref float speedX, ref float speedY, ref int type, ref int damage, ref float knockBack)
+		{
+			position = Main.MouseWorld;
+			return true;
+		}
```

Comment:
It should look like this (please test after and adjust pushYUp so it fits the sentry):
```suggestion
		public override bool Shoot(Player player, ref Vector2 position, ref float speedX, ref float speedY, ref int type, ref int damage, ref float knockBack) {
			player.FindSentryRestingSpot(type, out int worldX3, out int worldY3, out int pushYUp3); //This returns the coordinates for the sentry to be placed on solid ground below the cursor position
			//adjust pushYUp here depending on how high your projectile hitbox is
			Projectile.NewProjectile(worldX3, worldY3 - pushYUp3, 0f, 0f, type, damage, knockBack, Main.myPlayer);
			player.UpdateMaxTurrets(); //This despawns old sentries so spawning new ones will not go above player.maxTurrets
			return false;
		}
```

In the process, `UpdateMaxTurrets` should be removed from the projectile code.